### ISSUE TITLE
guided

### ISSUE THREAD

**Issue by awsssix** [huangqianfirst/Free_from-Image-Inpainting-with-Gated-Convolution#1]
@huangqianfirst Hi, Where could  get HED model?

**Comment by huangqianfirst** [comment]
I use thise code : https://github.com/meteorshowers/RCF-pytorch.

**Comment by awsssix** [comment]
@huangqianfirst you used pytorch?

**Comment by huangqianfirst** [comment]
Yes. But only used to pre-produce sketch image.

**Comment by erhuodaosi** [comment]
@huangqianfirst 
Have you finished it yet?
I have some questions about deepfillv2.
Q1:In the coarse-to-fine network,why we set l1_alpha?Is it a parameter to balance l1 loss and ae_loss?As a result,in deepfillv2 paper,should we set the value of l1_alpha in order to get mean l1 loss?
Q2:In DeepFillV2,the hingle loss is related to https://github.com/pfnet-research/sngan_projection/blob/master/updater.py.But I am confused about the hingle loss.
 I still could not understand the meaning of Ld and Lg.Could you explain it?
Q3:Why we use sketch channel(user guide)?Given a broken picture,if we use HED to generate the edge flist,and training it with edge flist.That's equal to know the broken area in advance.So what is the meaning to use sketch channel?By changing the broken area in a user guide way to generate the picture we want?How do you do in this sketch channel?Do you just put the original picture,mask and edge flist as input and train?
Q4:I have used free-form mask to generate mask which shape is (batchsize,1,256,256),I have set batchsize to 1,and I use batch_postprocess_images to reshpe the size to(256,256,1),the type is 'numpy.ndarray',How to print the mask?I have used in Image.formarray but in vain,maybe it is related to the number of channel numbers.

I would very appreciate if you could help me,thank you very much!
Best wishes to you!
